### PR TITLE
feat(data-types): CIDR, INET, MACADDR support for Postgres

### LINKED ISSUE #8154: Postgres support for cidr, init, macaddr column types
## What you are doing?

Trying to make use of the Postgres [network column types](https://www.postgresql.org/docs/current/static/datatype-net-types.html).

## What do you expect to happen?

That Sequelize would have DataType information for `CIDR`, `INET`, and `MACADDR` like it does for things like `UUID` or `HSTORE`.

## What is actually happening?

Sequelize currently has no idea what these columns mean. I've started working on a [branch](https://github.com/tadman/sequelize/tree/with_inet) to address this, but Issue #8153 is holding me up on testing it.

__Dialect:__ postgres
__Database version:__ 9.6
__Sequelize version:__ master

### PR DESCRIPTION
**Pull request by gabegorelick**
### Pull Request check-list

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](../CONTRIBUTING.md)?

### Description of change

Adds support for [Postgres network address types](https://www.postgresql.org/docs/9.6/static/datatype-net-types.html). 

This is a continuation of work started by @tadman in https://github.com/sequelize/sequelize/issues/8154.

Closes #8154

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "semver": "^5.5.0",
     "toposort-class": "^1.0.1",
     "uuid": "^3.2.1",
-    "validator": "^10.0.0",
+    "validator": "^10.4.0",
```
Comment (gabegorelick) [inline package.json]: Needed for `isIPRange`.

**Comment by gabegorelick** [comment]
Tests are now passing (the appveyor failure looks to be an existing flaky test). Let me know if more tests make sense. I'll also try to add some relevant docs to `model-definitions.md`.

File: lib/dialects/postgres/data-types.js [hunk 2]
```diff
@@ -647,6 +647,51 @@ module.exports = BaseTypes => {
     array_oids: []
   };
 
+  function CIDR() {
+    if (!(this instanceof CIDR)) return new CIDR();
+    BaseTypes.CIDR.apply(this, arguments);
+  }
+  inherits(CIDR, BaseTypes.CIDR);
+
+  CIDR.parse = function parse(value) {
```
Comment (sushantdhiman) [inline lib/dialects/postgres/data-types.js]: No need to add parse if we are not modifying this value, I think just defining oids should be enough for all three of these types

Comment (gabegorelick) [reply lib/dialects/postgres/data-types.js]: @sushantdhiman Done.

File: lib/dialects/postgres/data-types.js [hunk 3]
```diff
@@ -647,6 +647,39 @@ module.exports = BaseTypes => {
     array_oids: []
   };
 
+  function CIDR() {
```
Comment (sushantdhiman) [inline lib/dialects/postgres/data-types.js]: Even these function definitions are not required, as there is no special handling required for Postgres initializing of these datatypes.

Just define OIDs for these types at the top with others

Comment (gabegorelick) [reply lib/dialects/postgres/data-types.js]: Done.

**Comment by gabegorelick** [comment]
@sushantdhiman Thanks for shepherding this. Any chance it could get merged into v4?

**Comment by sushantdhiman** [comment]
@gabegorelick Please open a new PR targeting v4

**Comment by gabegorelick** [comment]
Thanks @sushantdhiman. I opened https://github.com/sequelize/sequelize/pull/9571 for that.

**Pull request merged**